### PR TITLE
Update windows instructions to encourage VS 2019 preview explicitly 

### PR DESCRIPTION
**Pull request by zsd4yr**

### CODE REVIEW

**Comment by GrabYourPitchforks** [comment]
Is VS2019 Preview actually _required_ going forward?

**Comment by Pilchie** [comment]
@GrabYourPitchforks I'm surprised that it isn't already, as you presumably need .NET Core 3 to build, and .NET Core 3 has required VS2019 for a while now.  It doesn't technically require the Preview channel yet, but it is likely to at some point.

**Comment by jkotas** [comment]
> Is VS2019 Preview actually required going forward?

It is not require to build. It is required for decent editing experience.

>  you presumably need .NET Core 3 to build

.NET Core build downloads its own local copy of the SDK. It does not require globally installed SDK.

**Comment by tannergooding** [comment]
> It is not require to build. It is required for decent editing experience.

We should likely remove it from the list of "required software" on Windows then; and list it as recommended instead (same goes for the .NET Core install requirement).

**Comment by zsd4yr** [comment]
> > It is not require to build. It is required for decent editing experience.
> 
> We should likely remove it from the list of "required software" on Windows then; and list it as recommended instead (same goes for the .NET Core install requirement).

I'm satisfied to put it where-ever we think it should go. 😄 

For the WinForms contribution guidelines, we were linking to this doc, and WinForms requires .NET 3.0, which I believe means we require VS 2019. Perhaps that means we should not link to this doc.

File: Documentation/building/windows-instructions.md [hunk 1]
```diff
@@ -4,8 +4,11 @@ Building CoreFX on Windows
...
-2. **[.NET Core SDK](https://www.microsoft.com/net/download/windows)** >= v3.0.0-preview3 must be installed which will add the `dotnet` toolchain to your path.
-3. **[CMake](https://cmake.org/)** must be installed from [the CMake download page](https://cmake.org/download/#latest) and added to your path. CMake 3.14 or later is required to build with VS 2019.
+1. **[CMake](https://cmake.org/)** must be installed from [the CMake download page](https://cmake.org/download/#latest) and added to your path. CMake 3.14 or later is required to build with VS 2019.
+
+## Recommended Software
+1. **[Visual Studio 2019 Preview](https://visualstudio.microsoft.com/vs/preview/)** (Community, Professional, Enterprise) with the latest update should be installed. The Community version is completely free.
+1. **[.NET Core SDK](https://www.microsoft.com/net/download/windows)** >= v3.0.0-preview3 should be installed, which will add the `dotnet` toolchain to your path.
```
Comment (zsd4yr) [inline Documentation/building/windows-instructions.md]: While we're here, this is the wrong link to send someone to if you want them to get a preview 3 SDK.

Only RTM full-release SDKs are available here (example: .NET Core 2.2); perhaps we should redirect to https://github.com/dotnet/core-sdk/#installers-and-binaries ?

Comment (danmosemsft) [reply Documentation/building/windows-instructions.md]: Why preview 3 specifically and not 2 or 4?

Comment (zsd4yr) [reply Documentation/building/windows-instructions.md]: @ViktorHofer added this according to the git blame

Comment (danmosemsft) [reply Documentation/building/windows-instructions.md]: Oh, I missed the `>=` bit.

**Comment by danmosemsft** [comment]
Thanks @zsd4yr

**Comment by ViktorHofer** [comment]
Thanks, changes look good!

**Pull request merged**